src/PubSubClient.cpp

```diff
@@ -655,16 +655,43 @@ void PubSubClient::disconnect() {
     pingOutstanding = false;
 }
 
+/**
+ * @brief  Write an UTF-8 encoded string to the give buffer and position. The string can have a length of 0 to 65535 bytes. The buffer is prefixed with two
+ * bytes representing the length of the string. See section 1.5.3 of MQTT v3.1.1 protocol specification.
+ * @note   If the string does not fit in the buffer (bufferSize) or is longer than 65535 bytes nothing is written to the buffer and the returned position
+ * is unchanged.
+ *
+ * @param  string 'C' string of the data that shall be written in the buffer.
+ * @param  buf Buffer to write the string into.
+ * @param  pos Position in the buffer to write the string.
+ * @return New position in the buffer (pos + 2 + string length), or pos if a buffer overrun would occur.
+ */
 size_t PubSubClient::writeString(const char* string, uint8_t* buf, size_t pos) {
-    const char* idp = string;
-    uint16_t i = 0;
-    pos += 2;
-    while (*idp) {
-        buf[pos++] = (uint8_t)*idp++;
-        i++;
+    return writeString(string, buf, pos, this->bufferSize);
+}
+
+/**
+ * @brief  Write an UTF-8 encoded string to the give buffer and position. The string can have a length of 0 to 65535 bytes. The buffer is prefixed with two
+ * bytes representing the length of the string. See section 1.5.3 of MQTT v3.1.1 protocol specification.
+ * @note   If the string does not fit in the buffer or is longer than 65535 bytes nothing is written to the buffer and the returned position is unchanged.
+ *
+ * @param  string 'C' string of the data that shall be written in the buffer.
+ * @param  buf Buffer to write the string into.
+ * @param  pos Position in the buffer to write the string.
+ * @param  size Maximal size of the buffer.
+ * @return New position in the buffer (pos + 2 + string length), or pos if a buffer overrun would occur.
+ */
+size_t PubSubClient::writeString(const char* string, uint8_t* buf, size_t pos, size_t size) {
+    size_t sLen = strlen(string);
```

Comment:
This should check for `string` not being a `nullptr`.
Otherwise `strlen` would throw an exception.

Reply:
Yes, this is a good point!
BTW. Do you know the
```
```suggestion
```
syntax to propose a concrete change?
https://stackoverflow.com/questions/54239733/how-can-i-manually-add-suggestions-in-code-reviews-on-github